mode/jinja2/jinja2.js

```diff
@@ -78,7 +78,24 @@
           state.instring = ch;
           stream.next();
           return "string";
-        } else if(stream.match(state.intag + "}") || stream.eat("-") && stream.match(state.intag + "}")) {
+        }
+        else if (state.inbraces > 0 && ch ==")") {
```

Comment:
Can you make sure `inbraces`/`inbrackets` are initialized in the start state?

Reply:
`inbraces / inbrackets` are accessed only in `intag` state.
Currently they are initialized right after the turning `intag` state on. I assume, this should be sufficient.

Reply:
It seems dodgy to compare `undefined > 0` here (though it will in fact produce false)